src/foundry/client/pixi/placeables/region.d.mts

```diff
@@ -0,0 +1,272 @@
+import type { ConfiguredObjectClassOrDefault } from "../../config.d.mts";
+import type { FixedInstanceType } from "../../../../utils/index.d.mts";
+import type RegionShape from "../../../client-esm/canvas/regions/shape.d.mts";
+import type RegionPolygonTree from "../../../client-esm/canvas/regions/polygon-tree.d.mts";
+import type RegionGeometry from "../../../client-esm/canvas/regions/geometry.d.mts";
+import type RegionMesh from "../../../client-esm/canvas/regions/mesh.d.mts";
+import type { Point } from "../../../common/types.d.mts";
+
+declare global {
+  namespace Region {
+    type ConfiguredClass = ConfiguredObjectClassOrDefault<typeof Region>;
+    type ConfiguredInstance = FixedInstanceType<ConfiguredClass>;
+
+    interface RenderFlags extends PlaceableObject.RenderFlags { }
+
+    interface RegionMovementWaypoint {
+      /** The x-coordinates in pixels (integer) */
+      x: number;
+      /** The y-coordinates in pixels (integer) */
+      y: number;
+      /** The elevation in grid units. */
+      elevation: number;
+    }
+
+    interface RegionMovementSegment {
+      /** The type of htis segment (see {@link Region.MOVEMENT_SEGMENT_TYPES}) */
+      type: number;
+      /** The waypoint that this segment starts from */
+      from: RegionMovementWaypoint;
+      /** The waypoint that this segment goes to */
+      to: RegionMovementWaypoint;
+    }
+  }
+
+  class Region extends PlaceableObject<RegionDocument.ConfiguredInstance> {
+    constructor(document: RegionDocument.ConfiguredInstance);
+
+    static override embeddedName: "Region";
+
+    static override RENDER_FLAGS: {
+      /** @defaultValue `{ propagate: ["refresh"] }` */
+      redraw: RenderFlag<Region.RenderFlags>;
+
+      /** @defaultValue `{ propagate: ["refreshState", "refreshBorder"], alias: true }` */
+      refresh: RenderFlag<Region.RenderFlags>;
+
+      /** @defaultValue `{}` */
+      refreshState: RenderFlag<Region.RenderFlags>;
+
+      /** @defaultValue `{}` */
+      refreshBorder: RenderFlag<Region.RenderFlags>;
+    }
+
+    // TODO(Eon): Figure out what and how to type the `static { ... }` block
```

Comment:
Can delete now that it's been explained.

To reiterate it has no type effects in of itself, it's just a fancy way to run code on class inclusion. In this case it does have effect due to `defineProperty` though.